File: projects/reference_nic/verif/test_nic_short/make_pkts.pl
#!/usr/local/bin/perl -w
# make_pkts.pl
#

use NF::PacketGen;
use NF::PacketLib;
use SimLib;

require reg_defines_reference_nic;

$delay = '@4us';
$batch = 0;
nf_set_environment( { PORT_MODE => 'PHYSICAL', MAX_PORTS => 4 } );

# use strict AFTER the $delay, $batch and %reg are declared
use strict;
use vars qw($delay $batch %reg);

# Prepare the DMA and enable interrupts
prepare_DMA('@3.9us');
enable_interrupts(0);

my $length = 100;
my $DA_sub = ':dd:dd:dd:dd:dd';
my $SA_sub = ':55:55:55:55:55';
my $DA;
my $SA;
my $pkt;
my $in_port;
my $out_port;
my $i = 0;
my $temp;

# send and receive 3 pkts into each port, should appear after DMA
$delay = '@17us';
$length = 60;
for($i=0; $i<12; $i++){
  $temp = sprintf("%02x", $i);
  $DA = $temp . $DA_sub;
  $SA = $temp . $SA_sub;
  $in_port = ($i%4) + 1;
  $pkt = make_IP_pkt($length, $DA, $SA, 64, '192.168.0.1', '192.168.0.2');
  nf_packet_in($in_port, $length, $delay, $batch,  $pkt);
  cpu_rxfifo_rd_pkt($in_port, $length, 0, $pkt);
  PCI_create_and_send_pkt($in_port, $length);
}

# check counter values
$delay='@120us';

# *********** Finishing Up - need this in all scripts ! ****************************
my $t = nf_write_sim_files();
print  "--- make_pkts.pl: Generated all configuration packets.\n";
printf "--- make_pkts.pl: Last packet enters system at approx %0d microseconds.\n",($t/1000);
if (nf_write_expected_files()) {
  die "Unable to write expected files\n";
}

nf_create_hardware_file('LITTLE_ENDIAN');
nf_write_hardware_file('LITTLE_ENDIAN');

File: projects/reference_nic/verif/test_nic_short/make_pkts.py
#!/usr/bin/python
# make_pkts.py

import testReg
import testPkt
import Test
import Packet

DA_sub = ':dd:dd:dd:dd:dd'; SA_sub = ':55:55:55:55:55';

Test.init()
testReg.regWrite(0x08, 0) #CPCI_Control_reg
testReg.regWrite(0x40, 0) #CPCI_Interrupt_Mask
testPkt.barrier()

length = 60;
for i in range(10):
  temp = "%02x" % i
  DA = temp + DA_sub
  SA = temp + SA_sub
  in_port = (i%4)+1
  pkt = Packet.make_IP_pkt(length, DA, SA, '64', '192.168.0.1', '192.168.0.2')
  print "- made IP pkt "
  if i%2 == 0:
    testPkt.pktSendPHY(in_port, pkt)
    testPkt.pktExpectDMA(in_port, pkt)
  else:
    testPkt.pktSendDMA(in_port, pkt)
    testPkt.pktExpectPHY(in_port, pkt)

  #testPkt.delay(1000)
  testReg.regWrite(0x0000020,0xabcd)
  testReg.regRead(0x0000020,0xabcd)
  testReg.regWrite(0x0000020,0xdead)
  testReg.regRead(0x0000020,0xdead)
  testReg.regWrite(0x0000020,0xabcd)
  testReg.regRead(0x0000020,0xabcd)
  testReg.regWrite(0x0000020,0xdead)
  testReg.regRead(0x0000020,0xdead)
  testReg.regWrite(0x0000020,0xabcd)
  testReg.regRead(0x0000020,0xabcd)
  testReg.regWrite(0x0000020,0xdead)
  testReg.regRead(0x0000020,0xdead)
  #testPkt.delay(1000)

#testPkt.delay(1000)
testPkt.barrier()

print  "--- make_pkts.py: Generated and sent all configuration packets.\n"
Test.close()
#Test.modelSim("eric")
